pymongo/operations.py

```diff
@@ -399,14 +542,16 @@ def __init__(
         self,
         filter: Mapping[str, Any],
         update: Union[Mapping[str, Any], _Pipeline],
-        upsert: bool = False,
+        upsert: Optional[bool] = None,
```

Comment:
I made the change since the client.bulk_write tests expect `upsert` to be optional and only passed into the command doc if it's set by the user. whereas those for coll.bulk_write expect `upsert` to always be passed into the command doc (by default as false)

Reply:
Oh I see, this is related to DRIVERS-604 which we never fully implemented.

Reply:
Opened https://jira.mongodb.org/browse/PYTHON-4608.